lib/model/message_list.dart

```diff
@@ -613,6 +619,15 @@ class MessageListView with ChangeNotifier, _MessageSequence {
   Anchor get anchor => _anchor;
   Anchor _anchor;
 
+  @override bool get oneMessagePerBlock => switch (narrow) {
+    CombinedFeedNarrow()
+      || ChannelNarrow()
+      || TopicNarrow()
+      || DmNarrow() => false,
+    MentionsNarrow()
+      || StarredMessagesNarrow() => true,
```

Comment:
This logic should get some dartdoc, to guide people adding cases to it in the future.